linode/helper/resource_datasource_config.go

```diff
@@ -25,3 +26,23 @@ func GetMetaFromProviderData(
 
 	return meta
 }
+
+func GetMetaFromProviderDataDatasource(
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) *FrameworkProviderMeta {
+	meta, ok := req.ProviderData.(*FrameworkProviderMeta)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf(
+				"Expected *http.Client, got: %T. Please report this issue to the provider developers.",
+				req.ProviderData,
+			),
+		)
+		return nil
+	}
+
+	return meta
+}
```

Comment:
Can we reuse `GetMetaFromProviderData` function? I can't see what's the difference